src/server/routes/response-handler.ts

```diff
@@ -0,0 +1,298 @@
+            );
+        }
+    }
+    res.status(500).end();
+});
+
+router.post(
+    "/getFormsBySession",
+    asyncHandler<
+        { MC: Array<Array<string>>; SA: Array<Array<string>> },
+        {},
+        { sid: string }
+    >(async (req, res) => {
+        const sid = req.body.sid;
+        const MCFormQuery = await MultipleChoiceResponseForm.find({
+            sessionID: sid,
+        });
+        const SAFormQuery = await ShortAnswerResponseForm.find({
+            sessionID: sid,
+        });
+
+        const MCForms = MCFormQuery.map((x) => x.id);
+        const SAForms = SAFormQuery.map((x) => x.id);
+        const MCFormQuestions = MCFormQuery.map((x) => x.question);
+        const SAFormQuestions = SAFormQuery.map((x) => x.question);
+
+        let MCFormData: Map<string, string> = new Map();
+        let SAFormData: Map<string, string> = new Map();
+
+        MCForms.forEach((key, i) => MCFormData.set(key, MCFormQuestions[i]));
+        SAForms.forEach((key, i) => SAFormData.set(key, SAFormQuestions[i]));
+
+        res.send({ MC: Array.from(MCFormData), SA: Array.from(SAFormData) })
+            .status(200)
+            .end();
+    })
+);
+
+router.post(
+    "/getMCFormByID",
+    asyncHandler<{ [optionID: string]: string }, {}, { formID: string }>(
+        async (req, res) => {
+            const query = await MultipleChoiceResponseForm.findById(
+                req.body.formID
+            );
+            if (query?.options) {
+                res.json(Object.fromEntries(query.options)).status(200).end();
+                return;
+            }
+            res.status(500).end();
+        }
+    )
+);
+
+router.post(
+    "/checkAnswered",
+    asyncHandler<{ found: boolean }, {}, { userID: string; formID: string }>(
+        async (req, res) => {
+            const query = await MultipleChoiceResponseForm.findById(
+                req.body.formID
+            );
+            if (query) {
+                if (query.answered.includes(req.body.userID)) {
+                    res.send({ found: true }).status(200).end();
+                    return;
+                }
+            } else {
+                const query2 = await ShortAnswerResponseForm.findById(
+                    req.body.formID
+                );
+                if (query2) {
+                    if (query2.answered.includes(req.body.userID)) {
+                        res.send({ found: true }).status(200).end();
+                        return;
+                    }
+                }
+            }
+            res.send({ found: false }).status(200).end();
+        }
+    )
+);
+
+router.post(
+    "/answerMultipleChoice",
+    asyncHandler<
+        undefined,
+        {},
+        { formID: string; userID: string; option: string }
+    >(async (req, res) => {
+        const options = new Map<string, string>(Object.entries(req.body));
+        const formID = options.get("formID");
```

Comment:
Here you've specified the type for res.body, so why is it converted to an object?